dev/com.ibm.ws.threading/src/com/ibm/ws/threading/internal/ThreadPoolController.java

```diff
@@ -673,7 +682,7 @@ enum LastAction {
         this.currentMinimumPoolSize = this.coreThreads;
         this.maxThreads = pool.getMaximumPoolSize();
         this.threadRange = this.maxThreads - this.coreThreads;
-        setPoolSize(coreThreads);
+        setPoolSize(startupPoolSize);
```

Comment:
Should there be any attempt to respect `maxThreads` during startup as well ? e.g. should `startupPoolSize` be allowed to override that potentially during startup (if it was a larger value than `maxThreads` that someone had for whatever reason) ?

Reply:
That's a good catch, thanks. I'll change that to 
`        setPoolSize(Math.min(startupPoolSize, this.maxThreads));`

Looking at that made me realize another issue ... 

Currently the thread pool controller does not run after the initial cycle if  `coreThreads == maxThreads` because there is nothing for it to do. But the addition of `startupPoolSize` means that the controller needs to run on its usual cycle until "ready to run", or it will never switch from `startupPoolSize` to `coreThreads`, so I need to change that also.

Reply:
OK, I've pushed an update to the PR that should fix both of the items mentioned above.

Reply:
Thanks Gary, the update looks fine to me.